### ISSUE TITLE
Missing Py_SetProgramName, Py_Initialize, PySys_SetArgv*

### ISSUE THREAD

**Issue by gitlab-importer** [pypy/pypy#3286]
In Heptapod by @SleepyRoy on Aug 23, 2020, 09:12

I just tried nightly (win32) binary 20200818 and good to see more data science modules such as pandas now work on pypy3. You know data science with Python is very popular so I believe this would greatly facilitate spread of pypy.

I tried to install jupyter notebook, one of the most popular IDEs. The other parts work nicely except for one dependency, `pywin32`, which results in jupyter kernel error (`cannot import win32api`). There's no official pywin32 whl for pypy so I copy pywin32 files to pypy's `site-packages` folder, just like how this module is installed on CPython. But this only enables `win32` and `win32con` imports, not `win32api` and `win32com`. Installing pypiwin32 also doesn't help at all.

I notice that someone has investigated this issue in https://github.com/mhammond/pywin32/issues/1289 and might help - 
*Main problem is that PyPy CPython compatibility C header Python.h(and PyPy library) does not provide these functions:
Py_SetProgramName, Py_Initialize, PySys_SetArgv*

Jupyter is not the only popular module depending on `pywin32`. I hope pypy could support it so more modules are enabled in pypy :)

**Comment by Avasam** [comment]
Also of interest that `Py_SetProgramName`,  `PySys_SetArgv` and `PySys_SetArgvEx` are deprecated:
- https://docs.python.org/3.15/c-api/interp-lifecycle.html#c.Py_SetProgramName
- https://docs.python.org/3.15/c-api/interp-lifecycle.html#c.PySys_SetArgv
- https://docs.python.org/3.15/c-api/interp-lifecycle.html#c.PySys_SetArgvEx

although it says "removed in Python 3.15", they're still available.

**Comment by StanFromIreland** [comment]
Indeed, they missed the feature-freeze. I have https://github.com/python/cpython/pull/149594 to move them to 3.16.
